bears/documentation/DocumentationStyleBear.py

```diff
@@ -46,75 +110,29 @@ def run(self, filename, file, language: str,
 
-            if main_description.desc == '\n' and not allow_missing_func_desc:
-                warning_desc = """
-Missing function description.
-Please set allow_missing_func_desc = True to ignore this warning.
-"""
-            else:
-                warning_desc = 'Documentation does not have correct style.'
-
-            # one empty line shall follow main description (except it's empty
-            # or no annotations follow).
-            if main_description.desc.strip() != '':
-                main_description = main_description._replace(
-                    desc='\n' + main_description.desc.strip() + '\n' *
-                         (1 if len(parsed) == 1 else 2))
-
-            new_metadata = [main_description]
-            for m in metadata:
-                # Split newlines and remove leading and trailing whitespaces.
-                stripped_desc = list(map(str.strip, m.desc.splitlines()))
-
-                if len(stripped_desc) == 0:
-                    # Annotations should be on their own line, though no
-                    # further description follows.
-                    stripped_desc.append('')
-                else:
-                    # Wrap parameter description onto next line if it follows
-                    # annotation directly.
-                    if stripped_desc[0] != '':
-                        stripped_desc.insert(0, '')
-
-                # Indent with 4 spaces.
-                stripped_desc = ('' if line == '' else ' ' * indent_size
-                                 + line for line in stripped_desc)
-
-                new_desc = '\n'.join(stripped_desc)
-
-                # Strip away trailing whitespaces and obsolete newlines (except
-                # one newline which is mandatory).
-                new_desc = new_desc.rstrip() + '\n'
-
-                new_metadata.append(m._replace(desc=new_desc.lstrip(' ')))
+        for doc_comment in self.extract(file, language, docstyle):
+            parsed = doc_comment.parse()
+
+            (new_metadata, warning_desc) = self.fix_parsed_data(
+                                parsed, allow_missing_func_desc, indent_size)
 
             new_comment = DocumentationComment.from_metadata(
                 new_metadata, doc_comment.docstyle_definition,
                 doc_comment.marker, doc_comment.indent, doc_comment.position)
 
             if new_comment != doc_comment:
                 # Something changed, let's apply a result.
-                diff = Diff(file)
-
-                # We need to update old comment positions, as `assemble()`
-                # prepends indentation for first line.
-                old_range = TextRange.from_values(
-                    doc_comment.range.start.line,
-                    1,
-                    doc_comment.range.end.line,
-                    doc_comment.range.end.column)
-                diff.replace(old_range, new_comment.assemble())
+                (affected_code, diffs) = self.generate_diff(
```

Comment:
As I commented in the other PR, accept the plain `Diff` object from the `generate_diff` method and assemble the `{"filename": <Diff>}` object here in this method... makes it clearer what's going on